gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+            "n_alt_alleles": t.info.n_alt_alleles,
+        },
+        50,
+        140,
+        extra_filter_checks=new_filters_dict,
+    )
+
+
+def histograms_sanity_check(
+    t: Union[hl.MatrixTable, hl.Table], verbose: bool, hists: List[str] = HISTS
+) -> None:
+    """
+    Check the number of variants that have nonzero values in their n_smaller and n_larger bins of quality histograms (both raw and adj).
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param List[str] hists: List of variant annotation histograms.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    for hist in hists:
+        for suffix in ["", "raw"]:
+            if suffix == "raw":
+                logger.info("Checking raw qual hists...")
+                hist = f"{hist}_{suffix}"
+            else:
+                logger.info("Checking adj qual hists...")
+
+            # Check subfield == 0
+            generic_field_check(
+                t,
+                cond_expr=(t.info[f"{hist}_n_smaller"] != 0),
+                check_description=f"{hist}_n_smaller == 0",
+                display_fields=[f"info.{hist}_n_smaller"],
+                verbose=verbose,
+            )
+            if hist not in [
+                "dp_hist_alt",
+                "dp_hist_all",
+            ]:  # NOTE: DP hists can have nonzero values in n_larger bin
+                generic_field_check(
+                    t,
+                    cond_expr=(t.info[f"{hist}_n_larger"] != 0),
+                    check_description=f"{hist}_n_larger == 0",
+                    display_fields=[f"info.{hist}_n_larger"],
+                    verbose=verbose,
+                )
+
+
+def raw_and_adj_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    subsets: List[str],
+    verbose: bool,
+    delimiter: str = "-",
+):
+    """
+    Perform sanity checks on raw and adj data in input Table.
+
+    Check that:
+        - Raw AC, AN, AF are not 0
+        - Adj AN is not 0 and AC and AF are not negative
+        - Raw values for AC, AN, nhomalt in each sample subset are greater than or equal to their corresponding adj values
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table to check.
+    :param List[str] subsets: List of sample subsets.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    for subfield in ["AC", "AF"]:
+        field = f"{subfield}{delimiter}"
```

Comment:
hmm, I guess we have used field a lot to refer to a hail table/struct variable that isn't an expression, so I support keeping it's use. I think here I was a bit confused because it changes the subfield to field, so maybe just 
```
for field in ["AC", "AF"]:
    field = f"{field}{delimiter}"
```
?